staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -140,6 +148,10 @@ func (a *Verifier) AuthenticateRequest(req *http.Request) (user.Info, bool, erro
 	if req.TLS == nil || len(req.TLS.PeerCertificates) == 0 {
 		return nil, false, nil
 	}
+	// do not verify cert and subjects of a requests that can not be handled by the wrapped auth.
+	if !a.auth.CanAuthenticateRequest(req) {
```

Comment:
this means the same could be used as an auth proxy cert (if user info headers are present) or as a client cert directly (if user info headers are omitted). That makes it too easy to accidentally authenticate as the wrong subject, which is why this was not allowed previously

Reply:
Not quite, if the request contains proxy headers then the x509 cert verify and common name checks are still performed first before the request header authentication.  It only only skips over the x509 verify and common namecheck for requests that do not contain any proxy authentication headers (which makes sense as there is no way that this authenticator is going to be able satisfy the request - it just falls through sooner, rather than validating the x509 only to fall through when it then tries to authenticate the wrapped request).

Reply:
are you using the same CA bundle as the `--requestheader-client-ca-file` and `--client-ca-file`?

Reply:
Hi - yes I think we do have the same CA bundle in use!  Can you point me at any more information on this?